app-nexus-network/app-nexus-network-htb-exports.js

```diff
@@ -0,0 +1,22 @@
+/**
+ *  This file contains any necessary functions that need to be exposed to the outside world.
+ *  Things like (render functions) will be exposed by adding them to the shellInterface variable, under the partners
+ *  profile name. This function will then be accessible through the window.headertag.AppNexusNetworkHtb object.
+ *  If necessary for backwards compatibility with old creatives, you can also add things directly to the
+ *  window namespace here, but this is discouraged if it's not strictly needed.
+ */
+
+//? if(FEATURES.GPT_LINE_ITEMS) {
+  shellInterface.AppNexusNetworkHtb = {
+    render: SpaceCamp.services.RenderService.renderDfpAd.bind(null, 'AppNexusNetworkHtb')
+};
+
+/* Existing creatives use window.pbjs.renderApnxAd */
+window.pbjs = window.pbjs || {};
+window.pbjs.renderApnxAd = SpaceCamp.services.RenderService.renderDfpAd.bind(null, 'AppNexusNetworkHtb');
```

Comment:
It looks like both AppNexus and AppNexus Network set window.pbjs.renderApnxAd and since the two adapters are meant to be deployed together, it seems like one will always overwrite the other. We suspect that this will lead to discrepancies.

Explanation: As is, one adapter will end up bound to renderApnxAd (either AppNexusHtb or AppNexusNetworkHtb, whichever runs this line last), and when creatives from either adapter attempt to call render, they'll end up making the same call. This means that when the adapter that isn't bound wins, the render will use the wrong partnerId. For example if window.pbjs.renderApnxAd is bound with AppNexusNetworkHtb, but AppNexus wins and its creative calls window.pbjs.renderApnxAd, then renderDfpAd will be called with AppNexusNetworkHtb, and it'll attempt to render AppNexusNetworkHtb's ad.

Reply:
Hi @ix-certification 

Thanks for the heads-up and the explanation.  

I believe we'll want to make this a distinct value from the other adapter.  To confirm though, this value is something that's used in the creative code?  If so and with this change, there would have to be separate creatives for both AppNexus adapters - correct?

Reply:
Hi @jsnellbaker,

Yes, its used in the creative code so that's why the 2 adapters will have separate creative tags.